.github/workflows/maven-package.yml

```diff
@@ -0,0 +1,30 @@
+# This workflow will build a Java project with Maven, and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Maven build
+
+on:
+  push:
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Code checkout
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+
+      - name: Build
+        run: mvn --batch-mode --threads 2C -DskipTests package
+
+      - name: Test
+        run: mvn --batch-mode --threads 2C -Dmaven.test.failure.ignore=true test
```

Comment:
Actually, this was copied from an example that used JUnit parsing. In the example the parser step wouldn't run if the `mvn test` step failed, so it's required for the `test` step to have a 0 exit code. We're not parsing results here so this shouldn't be here.